### DISCUSSION TITLE
Getting Selected Data from AGgrid

### DISCUSSION THREAD

**Discussion by Slanderkin** [zauberzeug/nicegui#626] [Q&A] [answered]
I want to be able to get selected rows of a ui.aggrid. I am trying to use the api call of getSelectedRows but it only ever returns None.

I saw [this discussion ](https://github.com/zauberzeug/nicegui/discussions/291) for ui.tables and attempted to use the solution even though tables and aggrids are different but I couldn't get that to work.

I modified the aggrid example for selection and if you select any number of rows then press the get selected button I only ever get None as a return.
```Python
from nicegui import ui

grid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name', 'checkboxSelection': True},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
    'rowSelection': 'multiple',
}).classes('max-h-40')

def update():
    grid.options['rowData'][0]['age'] += 1
    grid.update()

ui.button('Update', on_click=update)
ui.button('Get Selected', on_click=lambda: ui.notify(grid.call_api_method('getSelectedRows')))

ui.run()
```

I feel like I am misunderstanding the api call format or just not quite making a simple connection, any thoughts on what I missing?

**Comment by pythonormen** [comment, marked as answer]
I did as in https://github.com/zauberzeug/nicegui/discussions/404#discussioncomment-5055707 although for some reason it only worked occasionally. Like every third attempt (rest None)

**Reply by Slanderkin** [reply]
Thanks for your response, that worked for me all the time - I didn't run into the same issue you did.

For anyone else who stumbles upon this it is:
```Python
#For multiple rows
async def get_rows(self):
        rows = await table.get_selected_rows()
        #Some processing here

#For a single row
async def get_row(self):
        rows = await table.get_selected_row()
        #Some processing here
```

**Reply by rodja** [reply]
Thank you very much @pythonormen. It would be great if you could add this to the additional demos in the official documentation: [https://nicegui.io/documentation/table](https://nicegui.io/documentation/table#table_with_expandable_rows).

See https://github.com/zauberzeug/nicegui/blob/main/CONTRIBUTING.md#additional-demos for a short description on how to do this.
